Source/UI/NewUI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -31,13 +32,40 @@ class DefaultPaymentViewModel: PaymentViewModel {
         }
     }
 
+    func updateBillingSummaryView() {
+        guard let data = billingFormData, paymentFormStyle?.editBillingSummary != nil
+        else { return }
+
+        var summaryValue = ""
+
+        // user name
+        updateSummaryValue(with: data.name, summaryValue: &summaryValue)
+
+        // address
+        if let address = billingFormData?.address {
+            updateSummaryValue(with: address.addressLine1, summaryValue: &summaryValue)
+            updateSummaryValue(with: address.addressLine2, summaryValue: &summaryValue)
+            updateSummaryValue(with: address.city, summaryValue: &summaryValue)
+            updateSummaryValue(with: address.state, summaryValue: &summaryValue)
+            updateSummaryValue(with: address.zip, summaryValue: &summaryValue)
+            updateSummaryValue(with: address.country?.name, summaryValue: &summaryValue)
+        }
+
+        // phone
+        if let phone = billingFormData?.phone {
+            updateSummaryValue(with: phone.number, summaryValue: &summaryValue, withNewLine: false)
+            paymentFormStyle?.editBillingSummary?.summary?.text = summaryValue
```

Comment:
I am confused, from this code it appears `summaryValue` keeps being modified, but only ever gets assigned if the billing form has a phone. Is that the intention, to only do something with the output if a phone exists?

Reply:
Nice catch!